File: backend/src/main/java/com/pranay/ems/service/PayrollService.java
package com.pranay.ems.service;

import com.pranay.ems.dto.request.PayrollRequest;
import com.pranay.ems.dto.response.PayrollResponse;

import java.util.List;

public interface PayrollService {

    PayrollResponse generatePayroll(PayrollRequest request);

    PayrollResponse getPayrollById(Long id);

    List<PayrollResponse> getAllPayrolls();

    List<PayrollResponse> getPayrollByEmployee(Long employeeId);

    List<PayrollResponse> getPayrollByMonth(Integer month);

    List<PayrollResponse> getPayrollByYear(Integer year);

    List<PayrollResponse> getPayrollByMonthAndYear(
            Integer month,
            Integer year);

    void deletePayroll(Long id);
}

File: frontend/src/services/payrollService.js
import api from "../api/axios";

/* ==============================
   Generate Payroll
================================ */

export const generatePayroll = async (payroll) => {

    const response = await api.post(
        "/api/v1/payroll",
        payroll
    );

    return response.data;

};

/* ==============================
   Get All Payrolls
================================ */

export const getAllPayrolls = async () => {

    const response = await api.get(
        "/api/v1/payroll"
    );

    return response.data;

};

/* ==============================
   Get Payroll By Employee
================================ */

export const getPayrollByEmployee = async (employeeId) => {

    const response = await api.get(
        `/api/v1/payroll/employee/${employeeId}`
    );

    return response.data;

};

/* ==============================
   Get Payroll By Month
================================ */

export const getPayrollByMonth = async (month) => {

    const response = await api.get(
        `/api/v1/payroll/month/${month}`
    );

    return response.data;

};

/* ==============================
   Get Payroll By Year
================================ */

export const getPayrollByYear = async (year) => {

    const response = await api.get(
        `/api/v1/payroll/year/${year}`
    );

    return response.data;

};

/* ==============================
   Delete Payroll
================================ */

export const deletePayroll = async (id) => {

    const response = await api.delete(
        `/api/v1/payroll/${id}`
    );

    return response.data;

};